### ISSUE TITLE
support postgis 3.0

### ISSUE THREAD

**Issue by 0x777** [hasura/graphql-engine#3415]
When running the current test framework against PostGIS 3.0 (using the docker image `hasura/postgres:12.1-1`), there seem to be errors along these lines:
```sql
ERROR:  LWGEOM_dwithin: Operation on mixed SRID geometries (Point, 0) != (Polygon, 4326)
ERROR:  crosses: Operation on mixed SRID geometries (Point, 0) != (LineString, 4326)
```
Also from PostGIS 3.0, `raster` support has to be explicitly enabled with `CREATE EXTENSION postgis_raster`.

**Comment by lexi-lambda** [comment]
For context, the PostGIS 3.0 release notes are here: https://postgis.net/2019/10/20/postgis-3.0.0/

**Comment by paf31** [comment]
One concrete difference between versions 2 and 3 which is relevant here:

On v2:

```
# select ST_SRID(ST_GeomFromGeoJSON('{"type":"Point", "coordinates": [1,2]}'));
 st_srid 
---------
    0
(1 row)
```

and on v3:

```
# select ST_SRID(ST_GeomFromGeoJSON('{"type":"Point", "coordinates": [1,2]}'));
 st_srid 
---------
    4326
(1 row)
```

**Comment by paf31** [comment]
Another difference in behavior is that v2 evaluates this to `true` and v3 to `false`:

```sql
SELECT ST_Intersects(ST_GeographyFromText('POINT(-43.23456 72.4567772)'), ST_GeographyFromText('SRID=4326;LINESTRING(-43.23456 72.4567,-43.23456 72.4568)'));
```

**Comment by paf31** [comment]
Do we want separate CI tasks for the various combinations?

- PG 11 + postgis 2.5.3
- PG 11 + postgis 3.0.0
- PG 12 + postgis 2.5.3
- PG 12 + postgis 3.0.0

Right now, I just have PG {9,10,11} + postgis 2.5.3 and a separate PG 12 + postgis 3.0.0 job.

**Comment by lexi-lambda** [comment]
I think having what we have now is probably fine, and we can reevaluate if we ever find a situation where not having the extra jobs missed an actual bug. For now, it seems unlikely to be a problem.

**Comment by lexi-lambda** [comment]
Closed by #3519.
